fe/fe-core/src/main/java/org/apache/doris/nereids/NereidsPlanner.java

```diff
@@ -170,8 +170,10 @@ public Plan plan(LogicalPlan plan, PhysicalProperties requireProperties, Explain
             }
             deriveStats();
 
-            if (statementContext.getConnectContext().getSessionVariable().isEnableDPHypOptimizer()) {
-                // TODO: use DPHyp according the number of join table
+            if (!statementContext.getConnectContext().getSessionVariable().isDisableJoinReorder()
+                    && statementContext.getConnectContext().getSessionVariable().isEnableDPHypOptimizer()
+                    && statementContext.getMaxNAryInnerJoin() > statementContext.getConnectContext()
+                    .getSessionVariable().getMaxTableCountUseCascadesJoinReorder()) {
                 dpHypOptimize();
             } else {
                 optimize();
```

Comment:
i think we need always run optimize, and use different rule set according to join size.
merge dpHypOptimize and optimize into one function, and do if in it